pkg/controller/master/backup/backup_metadata.go

```diff
@@ -132,18 +133,40 @@ func (h *MetadataHandler) syncVMBackup(target *settings.BackupTarget) error {
 		return err
 	}
 
+	var namespaceFolders []string
+	var requiredMovingFilePaths []string
 	for _, fileName := range fileNames {
-		backupMetadata, err := loadBackupMetadataInBackupTarget(filepath.Join(metadataFolderPath, fileName), bsDriver)
+		filePath := filepath.Join(metadataFolderPath, fileName)
+		if bsDriver.FileExists(filePath) {
+			requiredMovingFilePaths = append(requiredMovingFilePaths, filePath)
+		} else {
+			namespaceFolders = append(namespaceFolders, filePath)
+		}
+	}
+
+	// collect all vmbackup metadata file paths in vmbackupMetadataFilePaths
+	vmbackupMetadataFilePaths := make([]string, len(requiredMovingFilePaths))
+	copy(vmbackupMetadataFilePaths, requiredMovingFilePaths)
+	for _, namespaceFolder := range namespaceFolders {
+		fileNames, err := bsDriver.List(namespaceFolder)
 		if err != nil {
 			return err
 		}
-		if backupMetadata.Namespace == "" {
-			backupMetadata.Namespace = metav1.NamespaceDefault
-		}
-		if err := h.createVMBackupIfNotExist(*backupMetadata, target); err != nil {
-			return err
+
+		for _, fileName := range fileNames {
+			filePath := filepath.Join(namespaceFolder, fileName)
+			if bsDriver.FileExists(filePath) {
+				vmbackupMetadataFilePaths = append(vmbackupMetadataFilePaths, filePath)
+			}
 		}
 	}
+
+	if err = h.loadBackupMetadataAndCreateVMBackup(target, vmbackupMetadataFilePaths, bsDriver); err != nil {
+		return err
+	}
+	if err = h.moveFilePaths(requiredMovingFilePaths, bsDriver); err != nil {
+		return err
+	}
```

Comment:
If my understanding is correct, this would load metadata and create vmbackup CR first, then move existing metadata to the correct path in the backup target if needed ?

I'm considering if there is any power outage on the node or the backup target, and making incomplete metadata left on target. During next restart, would this bring conflict with vmbackup controller [handleBackupReady()](https://github.com/FrankYang0529/harvester/blob/bc7abe2d219b490b40b41736800326d3473ea293/pkg/controller/master/backup/backup.go#L979-L1000)?

Reply:
I think the situation is fine, because the vm metadata which is used to create VMBackup in a new cluster will be the same content in [uploadVMBackupMetadata](https://github.com/FrankYang0529/harvester/blob/bc7abe2d219b490b40b41736800326d3473ea293/pkg/controller/master/backup/backup.go#L999). Both `moveFilePaths` and `uploadVmBackupMetadata` are trying to save same content to a same path. The different order of `moveFilePaths` and `uploadVmBackupMetadata` will get same result.

Reply:
However, such interleaving seems avoidable. What if change the order to make the metadata operation sequential? Would it bring any issue?

Reply:
Do you mean that we only move files in `uploadVMBackupMetadata`?

Reply:
I mean like:
```
	if err = h.moveFilePaths(requiredMovingFilePaths, bsDriver); err != nil {
		return err
	}
	if err = h.loadBackupMetadataAndCreateVMBackup(target, vmbackupMetadataFilePaths, bsDriver); err != nil {
		return err
	}
```
Would it cost lots of effort? I think it may introduce some restructuring on `syncVMBackup`. Thanks.